src/main/scala/esmeta/compiler/Compiler.scala

```diff
@@ -443,7 +443,10 @@ class Compiler(
         ESubstring(
           compile(fb, expr),
           compile(fb, from),
-          compile(fb, to),
+          to match {
```

Comment:
Try to use the `map` function of the `Option[_]` type, or [For Comprehension](https://docs.scala-lang.org/tour/for-comprehensions.html). Do not use curly braces if possible.